### PR TITLE
Support adding a caption to media uploads.

### LINKED ISSUE #3508: [Task] Add a caption when sending a media file
Please see and discuss the details in the meta issue.

### PR DESCRIPTION
**Pull request by pixlwave**
This PR reuses the message composer to add captions to media files.

Closes #3508.

https://github.com/user-attachments/assets/c2e45018-ad27-4dcd-962f-8d6cbb2e8779

### CODE REVIEW

File: ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift [hunk 1]
```diff
@@ -118,3 +139,10 @@ class MediaUploadPreviewScreenViewModel: MediaUploadPreviewScreenViewModelType,
         userIndicatorController.submitIndicator(UserIndicator(title: label))
     }
 }
+
+extension NSAttributedString {
+    var nonEmptyString: String? {
```
Comment (Velin92) [inline ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift:144]: I would suggest no just checking if is just empty but also if is not blank, check the `isBlank` in `String` extension.

File: ElementX/Sources/Screens/RoomScreen/ComposerToolbar/View/MessageComposer.swift [hunk 2]
```diff
@@ -200,6 +188,27 @@ private struct MessageComposerHeaderLabelStyle: LabelStyle {
     }
 }
 
+extension View {
```
Comment (Velin92) [inline ElementX/Sources/Screens/RoomScreen/ComposerToolbar/View/MessageComposer.swift:191]: `private` ?

**Review by Velin92 — APPROVED** [review]

File: ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/MediaUploadPreviewScreen/MediaUploadPreviewScreenViewModel.swift]: Ah yeah nice, forgot about that, updated in https://github.com/element-hq/element-x-ios/pull/3531/commits/316321d5089ddffed08b2d1ddd5c4f2cd600c2c9

**Comment by pixlwave** [comment]
Going to wait until after the RC before merging this.

**Pull request merged**